src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs

```
@@ -1418,7 +1421,7 @@ private static string FormatTable(string table, string schema)
                 return;
             }
 
-            var columnType = operation.ColumnType ?? GetColumnType(schema, table, name, operation, model);
+            var columnType = operation.ColumnType ?? GetColumnType(schema, table, name, operation, model)!;
```

Do we want to throw an exception if the column type isn't found?

---

That would be an extreme corner case. The user would have to be adding an out-of-band table/column in the migration, specifying a CLR type the provider doesn't know about, *and* not specifying a SQL type.